MaraBot/Core/PresetValidation.cs

```diff
@@ -0,0 +1,185 @@
+                    if (comp == 0)
+                        errors.Add($"{VALIDATION_INFO_PREFIX} Min amount of mana seeds required is the same as the max amount. You should instead just set the 'Mana Tree Revival seeds required'.");
+                }
+
+                // Don't use the override christmas settings if already on a christmas goal
+                if (optionsCopy.ContainsKey("opGoal") && (optionsCopy["opGoal"] == "gift" || optionsCopy["opGoal"] == "reindeer"))
+                {
+                    if (optionsCopy.ContainsKey("opXmasMaps"))
+                        errors.Add($"{VALIDATION_INFO_PREFIX} Setting the Christmas theme explicitly is unnecessary, as the goal is already Christmas-themed.");
+                    if (optionsCopy.ContainsKey("opXmasItems"))
+                        errors.Add($"{VALIDATION_INFO_PREFIX} Setting the random Christmas drops explicitly is unnecessary, as the goal is already Christmas-themed.");
+                }
+
+                // Check for race-safety.
+                // Note that we check the actual options,
+                // because we need to know when opSpoilerLog
+                // is unparseable.
+                if (options.ContainsKey("opSpoilerLog") && options["opSpoilerLog"] == "no")
+                    errors.Add($"{VALIDATION_GOOD_PREFIX} Options are race-safe.");
+                else if (options.ContainsKey("opSpoilerLog"))
+                    errors.Add($"{VALIDATION_INFO_PREFIX} Options might not be race-safe, because I don't know if a spoiler log gets generated.");
+                else
+                    errors.Add($"{VALIDATION_INFO_PREFIX} Options are not race-safe, because a spoiler log gets generated.");
+            }
+
+            /*
+             * Ancient Cave
+             */
+            if (mode == Mode.AncientCave)
+            {
+				if (optionsCopy.ContainsKey("acBoy") && optionsCopy.ContainsKey("acGirl") && optionsCopy.ContainsKey("acSprite"))
+					errors.Add($"{VALIDATION_ERROR_PREFIX} Must have at least one character to start with.");
+				if (optionsCopy.ContainsKey("acBiomeTypes") && Option.ParseList(optionsCopy["acBiomeTypes"]).Count == 0)
+					errors.Add($"{VALIDATION_ERROR_PREFIX} Must have at least one floor type selected.");
+			}
+
+            /*
+             * Boss Rush
+             */
+            if (mode == Mode.BossRush)
+            {
+				if (optionsCopy.ContainsKey("brBoy") && optionsCopy.ContainsKey("brGirl") && optionsCopy.ContainsKey("brSprite"))
+					errors.Add($"{VALIDATION_ERROR_PREFIX} Must have at least one character to start with.");
+			}
+
+            /*
+             * Chaos
+             */
+            if (mode == Mode.Chaos)
+            {
+				if (optionsCopy.ContainsKey("chBoy") && optionsCopy.ContainsKey("chGirl") && optionsCopy.ContainsKey("chSprite"))
+					errors.Add($"{VALIDATION_ERROR_PREFIX} Must have at least one character to start with.");
+			}
```

Comment:
This code is using wrong indentations. There seems to be a mismatch of tabs/spaces in this file. We should use spaces. `editorconfig` failed us 🤔

Reply:
It probably failed because this is based on an old file that got edited outside of the project. Fixed the indentation.